.buildkite/shared-pipeline-vars

```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+# This file is `source`'d before calling `buildkite-agent pipeline upload`, and can be used
+# to set up some variables that will be interpolated in the `.yml` pipeline before uploading it.
+
+# The ~> modifier is not currently used, but we check for it just in case
+XCODE_VERSION=$(sed -E -n 's/^(~> )?(.*)/xcode-\2/p' .xcode-version)
+CI_TOOLKIT_PLUGIN_VERSION="3.5.1"
+NVM_PLUGIN_VERSION='0.3.0'
+
+export IMAGE_ID="$XCODE_VERSION"
```

Comment:
💡 given the variable name, I wonder if it wouldn't be clearer if `XCODE_VERSION` would only contain the version, and `IMAGE_ID` would be the one adding the prefix?

Would also make the `sed` simpler and easier to understand—as the pattern would now just tell "remove the `~> ` prefix if present" without the need for `?`, captures, nor backreferences 🙂

```suggestion
XCODE_VERSION=$(sed 's/^~> *//' .xcode-version)
CI_TOOLKIT_PLUGIN_VERSION="3.5.1"
NVM_PLUGIN_VERSION='0.3.0'

export IMAGE_ID="xcode-$XCODE_VERSION"
```